App.tsx

```diff
@@ -1,48 +1,58 @@
-import React, { useEffect, useState } from 'react';
+import React from 'react';
 import { StyleSheet, Text, View, Image } from 'react-native';
+import { NavigationContainer } from '@react-navigation/native';
+import { createStackNavigator } from '@react-navigation/stack';
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+import { Ionicons } from '@expo/vector-icons';
+import FavPokemon from './components/FavPokemon';
+import PokemonList from './components/PokemonList';
+import NativeModule from './components/NativeModule';
 
-export default function App() {
-  const [vulpix, setVulpix] = useState(null);
-  const [loading, setLoading] = useState(true);
-
-  const getVulpix = async () => {
-    try {
-      const res = await fetch('https://pokeapi.co/api/v2/pokemon/37');
-      const data = await res.json();
-
-      setVulpix(data);
-      setLoading(false);
-    } catch (error) {
-      console.error(error);
-    }
-  };
+const Stack = createStackNavigator();
+const TabNav = createBottomTabNavigator();
 
-  useEffect(() => {
-    getVulpix();
-    //eslint-disable-next-line
-  }, []);
+export default function App() {
+  return (
+    <NavigationContainer>
+      <TabNav.Navigator
+        screenOptions={({ route }) => ({
+          tabBarIcon: ({ focused, color, size }) => {
+            let iconName;
 
-  if (loading) {
-    return (
-      <View style={styles.container}>
-        <Text>Loading...</Text>
-      </View>
-    );
-  } else {
-    return (
-      <View style={styles.container}>
-        <Text style={styles.header}>This is {vulpix.name}!</Text>
-        <Image
-          style={styles.image}
-          source={{ uri: vulpix.sprites.front_default }}
+            if (route.name === 'FavPokemon') {
+              iconName = 'md-heart';
+            } else if (route.name === 'PokemonList') {
+              iconName = 'md-paw';
+            } else if ((route.name = 'NativeModule')) {
+              iconName = 'md-cafe';
+            }
+            return <Ionicons name={iconName} size={size} color={color} />;
+          }
+        })}
+        tabBarOptions={{
+          activeTintColor: '#59B6AE',
+          inactiveTintColor: 'gray'
+        }}
+      >
+        <TabNav.Screen name='FavPokemon' component={FavPokemon} />
```

Comment:
TypeScript allows devs to manage strings a bit more easily with enums. Instead of copy-pasting the same string here and above in `tabBarIcon`, you could create such an enum:
```ts
enum TabScreenNames {
  FAV_POKEMON = 'FavPokemon',
  ...
}
```
and then use it like normal objects with fields:
`if (route.name === TabScreenNames.FAV_POKEMON) {`
`<TabNav.Screen name={TabScreenNames.FAV_POKEMON} component={FavPokemon} />`